oauth.rst

```diff
@@ -1,131 +1,129 @@
 
-As a result of registering, clients are given a ``client_id`` and a ``client_secret`` that they can use to sign their requests to our api.
+Step 2 - Authorization page
+---------------------------
 
-State Variable
---------------
-Clients are encouraged to use the `state` variable to prevent CSRF attacks. A good state variable could be a self-signed string containing some simple info like:
+The User is presented with a page in the Dock application where they need to:
 
-- Current url to redirect the user to the right page once back
-- User_id
+i) Authenticate (if they haven't already done so)
+ii) Check Application info (name, description, logo, etc)
+iii) Approve the requested access to the given scopes
 
-The ``state`` var is returned unmodified back to the client by us, the Provider. So if they sign the ``state`` var properly, they can verify that it contains the right user id, and also a valid url to redirect to. If a CSRF attack took place, the signature will be broken and they can abort that authentication flow.
+Additionally, the user is presented the following items:
 
-Redirect URIs
--------------
-``dock-auth`` only accepts https addresses. That means that native apps are not supported for now. We'll use **exact match** only, to avoid acting as an open redirector, and it is forbidden to provide URLs containing anything after the fragment identifier.
+i) Detailed description of the scope (or which resources the client is requesting access to).
+ii) Link to application's Privacy Policy
+iii) Link to register if the user doesn't own an account at Dock
 
-Scopes
-------
-A ``scope`` is a way to limit a 3rd party app's access to a user's data. In our case we're taking a two-step approach:
-1 - First we'll only accept a "basic" scope that allows access to first- and last name, email, avatar and dock user id.
-2 - Once the ``dock-gateway`` is fully operational, we'll want to only share user data through the Dock protocol. For that, the "basic" scope will be modified to only contain the ETH address of the contract between the Client and the User. The Client can then use this address to go and interact with the contract directly in the Ethereum network, or ask the ``dock-gateway`` to fetch and decrypt the user data.
-Partners that implement their integration during our first step, will need to adjust their implementation once we move to step 2 since these will be breaking changes.
-TODO: change step 2 wording once dock-gateway is fully operational.
+Once the user clicks on "Authorize", we redirect him to the given ``redirect_uri`` with the following two parameters added to the url:
```

Comment:
User is sometimes "user" and some other times "User", should we choose one and try to make it consistent?